### PR TITLE
Discrepancy gradient as function of policy params

### PR DESCRIPTION
**Pull request by dsctt**

### CODE REVIEW

File: grl/grad.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import logging
+
+import numpy as np
+
+from jax import grad
+
+def do_grad(pe, pi_abs, no_gamma, lr=1):
```
Comment (camall3n) [inline grl/grad.py:7]: Can you change `pe` to `policy_eval` or something more verbose?

File: grl/grad.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    i = 0
+    done_count = 0
+    while done_count < 5:
+        i += 1
+        if i % 10 == 0:
+            print('Gradient iteration', i)
+
+        pi_grad = grad(mse_loss)(pi_abs)
+        old_pi = pi_abs
+        pi_abs -= lr * pi_grad
```
Comment (camall3n) [inline grl/grad.py:24]: Wow, Jax is amazing

File: grl/run.py [hunk 3]
```diff
@@ -9,8 +9,9 @@
 from .mdp import MDP, AbstractMDP
 from .mc import mc
 from .policy_eval import PolicyEval
+from .grad import do_grad
 
-def run_algos(spec, no_gamma, n_random_policies, n_steps, max_rollout_steps):
+def run_algos(spec, no_gamma, n_random_policies, grad, n_steps, max_rollout_steps):
```
Comment (camall3n) [inline grl/run.py:14]: Can you change `grad` to `use_grad` to avoid confusion with `jax.grad`?

**Review by camall3n — CHANGES_REQUESTED** [review]
This is pretty slick

File: grl/policy_eval.py [hunk 4]
```diff
@@ -40,14 +46,16 @@ def solve_mdp(self, mdp):
         b = [] # Each index will contain the sum of the constants for that equation   (list of floats)
         for s in range(mdp.n_states):
             a_t = np.zeros(mdp.n_states)
-            a_t[s] = -1 # subtract V_pi(s) to right side
+            # a_t[s] = -1 # subtract V_pi(s) to right side
+            a_t = a_t.at[s].set(-1)
```
[the remark this answers is not in the document]
Comment (dsctt) [reply grl/policy_eval.py:50]: Yea, jax arrays are immutable. Has to be done this way.

File: grl/grad.py (hunk 1 above)
Comment (dsctt) [reply grl/grad.py:7]: Done

File: grl/run.py (hunk 3 above)
Comment (dsctt) [reply grl/run.py:14]: Done

**Review by camall3n — APPROVED** [review]
